src/main/java/uk/ac/cam/cl/dtg/teaching/pottery/controllers/RepoController.java

```diff
@@ -37,55 +37,50 @@
 import uk.ac.cam.cl.dtg.teaching.pottery.exceptions.TaskNotFoundException;
 import uk.ac.cam.cl.dtg.teaching.pottery.model.FileData;
 import uk.ac.cam.cl.dtg.teaching.pottery.model.RepoInfo;
+import uk.ac.cam.cl.dtg.teaching.pottery.model.RepoInfoWithStatus;
 import uk.ac.cam.cl.dtg.teaching.pottery.model.RepoTag;
 import uk.ac.cam.cl.dtg.teaching.pottery.repo.Repo;
 import uk.ac.cam.cl.dtg.teaching.pottery.repo.RepoFactory;
 import uk.ac.cam.cl.dtg.teaching.pottery.task.Task;
 import uk.ac.cam.cl.dtg.teaching.pottery.task.TaskCopy;
 import uk.ac.cam.cl.dtg.teaching.pottery.task.TaskIndex;
+import uk.ac.cam.cl.dtg.teaching.pottery.worker.Job;
+import uk.ac.cam.cl.dtg.teaching.pottery.worker.Worker;
 
 public class RepoController implements uk.ac.cam.cl.dtg.teaching.pottery.api.RepoController {
 
   protected static final Logger LOG = LoggerFactory.getLogger(RepoController.class);
   private RepoFactory repoFactory;
   private TaskIndex taskIndex;
+  private Worker worker;
 
   /** Create a new RepoController. */
   @Inject
-  public RepoController(RepoFactory repoFactory, TaskIndex taskIndex) {
+  public RepoController(RepoFactory repoFactory, TaskIndex taskIndex, Worker worker) {
     super();
     this.repoFactory = repoFactory;
     this.taskIndex = taskIndex;
+    this.worker = worker;
   }
 
   @Override
-  public RepoInfo makeRemoteRepo(
+  public RepoInfoWithStatus makeRemoteRepo(
       String taskId,
-      Boolean usingTestingVersion,
-      Integer validityMinutes,
+      Boolean usingTestingVersionBoolean,
+      Integer validityMinutesInteger,
       String variant,
       String remote)
-      throws TaskNotFoundException, RepoExpiredException, RepoStorageException,
-          RetiredTaskException, RepoNotFoundException, TaskMissingVariantException {
+      throws TaskNotFoundException, RepoStorageException, RetiredTaskException,
+      RepoNotFoundException, TaskMissingVariantException {
     if (taskId == null) {
       throw new TaskNotFoundException("No taskId specified");
     }
-    if (usingTestingVersion == null) {
-      usingTestingVersion = false;
-    }
-    if (validityMinutes == null) {
-      validityMinutes = 60;
-    }
+    final boolean usingTestingVersion = usingTestingVersionBoolean == null ? false
```

Comment:
any reason these are explicitly final?

Reply:
Overzealous typing! A Java 8 feature I'm not used to. I'll take them out and keep it consistent.